### PR TITLE
[WFLY-19793] App client container should log java.lang.NoClassDefFoundError failures and other failures while loading the appclient main class

### PR DESCRIPTION
**Pull request by scottmarlow**
https://issues.redhat.com/browse/WFLY-19793

This should improve the logging of failures seen when loading the app client main class

### CODE REVIEW

File: appclient/src/main/java/org/jboss/as/appclient/deployment/ApplicationClientManifestProcessor.java [hunk 1]
```diff
@@ -56,12 +56,12 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
                         if (environment != null) {
                             DescriptorEnvironmentLifecycleMethodProcessor.handleMethods(environment, moduleDescription, mainClass);
                         }
-                    } catch (ClassNotFoundException e) {
+                    } catch (Throwable e) {
```
Comment (bstansberry) [inline appclient/src/main/java/org/jboss/as/appclient/deployment/ApplicationClientManifestProcessor.java:59]: This changes the error handling for L50-58.

Refactoring a bit avoids that:

```
                   final Class<?> clazz;
                    try {
                        clazz = module.getClassLoader().loadClass(mainClass);
                    } catch (Throwable t) {
                        throw AppClientLogger.ROOT_LOGGER.cannotLoadAppClientMainClass(t);
                    }
                   deploymentUnit.putAttachment(AppClientAttachments.MAIN_CLASS, clazz);
                   ...
```

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: appclient/src/main/java/org/jboss/as/appclient/deployment/ApplicationClientManifestProcessor.java (hunk 1 above)
Comment (scottmarlow) [reply appclient/src/main/java/org/jboss/as/appclient/deployment/ApplicationClientManifestProcessor.java:51]: Thanks, looks better!

**Review by bstansberry — APPROVED** [review]

File: appclient/src/main/java/org/jboss/as/appclient/deployment/ApplicationClientManifestProcessor.java [hunk 2]
```diff
@@ -45,23 +45,23 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
...
 
+                    final DeploymentDescriptorEnvironment environment = deploymentUnit.getAttachment(org.jboss.as.ee.component.Attachments.MODULE_DEPLOYMENT_DESCRIPTOR_ENVIRONMENT);
+                    if (environment != null) {
+                        DescriptorEnvironmentLifecycleMethodProcessor.handleMethods(environment, moduleDescription, mainClass);
+                    }
                 }
             }
         }
     }
-}
```
Comment (bstansberry) [inline appclient/src/main/java/org/jboss/as/appclient/deployment/ApplicationClientManifestProcessor.java:67]: @scottmarlow For this to compile this needs to be restored.

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: appclient/src/main/java/org/jboss/as/appclient/deployment/ApplicationClientManifestProcessor.java (hunk 2 above)
Comment (bstansberry) [reply appclient/src/main/java/org/jboss/as/appclient/deployment/ApplicationClientManifestProcessor.java:67]: I added a commit fixing this so we can get it merged.

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @scottmarlow

**Comment by scottmarlow** [comment]
Thanks @bstansberry!  Nice to see this change in!

**Pull request merged**